### DISCUSSION TITLE
Duplicate global variable names

### DISCUSSION THREAD

**Discussion by Jinxishihenian** [webpack/webpack#16423] [Q&A]
# Bug report

**What is the current behavior?**

After packaging the file, the global variable name is repeated, resulting in the variable is overwritten, call the wrong method, resulting in an error.

**If the current behavior is a bug, please provide the steps to reproduce.**
Execute the command “cross-env BUILD_ENV=dev umi build“

**What is the expected behavior?**
Global variable definitions do not affect each other

**Other relevant information:**
webpack version:  5.6.0
Node.js version: 16.14.2
Operating System: win 10
Additional tools:

**Related images**
![50718c6bf5d2faddaf77787838183d4](https://user-images.githubusercontent.com/44466363/198245311-6ec05b4c-9dfe-4f1f-ae65-1345c48df61d.png)
![916ebba92289fd14e9ea470e31587c7](https://user-images.githubusercontent.com/44466363/198245347-d90a7f59-184a-47d8-9752-835f8bf88309.png)

**Comment by alexander-akait** [comment]
We don't do it, do you have non official plugins?

**Reply by Jinxishihenian** [reply]
Yes, we use umiJS, which encapsulates the webpack configuration again using.umirc.ts
Configure the address:
https://v3.umijs.org/zh-CN/docs/directory-structure#umircts

**Reply by Jinxishihenian** [reply]
Find the cause, third-party plug-in problems caused by the.umirc configuration cause.

**Comment by lzw10168** [comment]
I had the same problem and worked on it for days, The esbuild configuration needs to be closed. 


![image](https://github.com/webpack/webpack/assets/67814702/1fda0049-d435-4c16-9632-996c7a217b27)

**Reply by lzw10168** [reply]
@Jinxishihenian
